File: benchmark/binary/x86-64/window/small/source/golang/test_functions.go
package main

import "fmt"

// Simple arithmetic functions
func add(a, b int) int {
	return a + b
}

func multiply(a, b int) int {
	return a * b
}

// Find maximum
func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

// Fibonacci with memoization
func fibonacci(n int) int {
	cache := make(map[int]int)
	var fib func(int) int
	fib = func(x int) int {
		if x <= 1 {
			return x
		}
		if val, ok := cache[x]; ok {
			return val
		}
		result := fib(x-1) + fib(x-2)
		cache[x] = result
		return result
	}
	return fib(n)
}

// Sum array
func sumArray(arr []int) int {
	sum := 0
	for _, val := range arr {
		sum += val
	}
	return sum
}

// Process code with switch
func processCode(code int) int {
	switch code {
	case 1:
		return 10
	case 2:
		return 20
	case 3:
		return 30
	default:
		return 0
	}
}

// Fill matrix
func fillMatrix(rows, cols, value int) [][]int {
	matrix := make([][]int, rows)
	for i := range matrix {
		matrix[i] = make([]int, cols)
		for j := range matrix[i] {
			matrix[i][j] = value
		}
	}
	return matrix
}

// Swap two values
func swap(a, b *int) {
	*a, *b = *b, *a
}

func main() {
	x := 5
	y := 10

	result1 := add(x, y)
	result2 := max(x, y)
	result3 := fibonacci(10)

	arr := []int{1, 2, 3, 4, 5}
	arrSum := sumArray(arr)

	codeResult := processCode(2)

	swap(&x, &y)

	matrix := fillMatrix(3, 3, 5)

	fmt.Printf("Results: %d %d %d %d %d\n", result1, result2, result3, arrSum, codeResult)
	fmt.Printf("Matrix size: %d x %d\n", len(matrix), len(matrix[0]))
}

File: benchmark/binary/x86-64/window/small/source/c/test_functions.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// Simple arithmetic function
int add(int a, int b) {
    return a + b;
}

// Function with control flow
int max(int a, int b) {
    if (a > b) {
        return a;
    }
    return b;
}

// Recursive function
int fibonacci(int n) {
    if (n <= 1) {
        return n;
    }
    return fibonacci(n - 1) + fibonacci(n - 2);
}

// Function with loop
int sum_array(int *arr, int len) {
    int sum = 0;
    for (int i = 0; i < len; i++) {
        sum += arr[i];
    }
    return sum;
}

// Function with switch
int process_code(int code) {
    switch (code) {
        case 1:
            return 10;
        case 2:
            return 20;
        case 3:
            return 30;
        default:
            return 0;
    }
}

// Function with nested loops
void fill_matrix(int *matrix, int rows, int cols, int value) {
    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < cols; j++) {
            matrix[i * cols + j] = value;
        }
    }
}

// Function with pointer manipulation
void swap(int *a, int *b) {
    int temp = *a;
    *a = *b;
    *b = temp;
}

// Main function
int main() {
    int x = 5;
    int y = 10;
    
    int result1 = add(x, y);
    int result2 = max(x, y);
    int result3 = fibonacci(10);
    
    int arr[] = {1, 2, 3, 4, 5};
    int arr_sum = sum_array(arr, 5);
    
    int code_result = process_code(2);
    
    swap(&x, &y);
    
    printf("Results: %d %d %d %d %d\n", result1, result2, result3, arr_sum, code_result);
    
    return 0;
}
